packages/typespec-azure-resource-manager/test/rules/arm-no-record.test.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  BasicTestRunner,
+  LinterRuleTester,
+  createLinterRuleTester,
+} from "@typespec/compiler/testing";
+import { beforeEach, describe, it } from "vitest";
+import { armNoRecordRule } from "../../src/rules/arm-no-record.js";
+import { createAzureResourceManagerTestRunner } from "../test-host.js";
+
+describe("typespec-azure-resource-manager: ARM no-record rule", () => {
+  let runner: BasicTestRunner;
+  let tester: LinterRuleTester;
+
+  beforeEach(async () => {
+    runner = await createAzureResourceManagerTestRunner();
+    tester = createLinterRuleTester(
+      runner,
+      armNoRecordRule,
+      "@azure-tools/typespec-azure-resource-manager"
+    );
+  });
+
+  it("emits diagnostic when a model property uses Record type", async () => {
+    await tester
+      .expect(
+        `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Contoso;
+
+      @Azure.ResourceManager.tenantResource
+      model Widget is ProxyResource<WidgetProperties> {
+        @key("widgetName")
+        @segment("widgets")
+        @path
+        @visibility("read")
+        name: string;
+
+        props: Record<string>;
+      }
+
+      model WidgetProperties {}
+      `
+      )
+      .toEmitDiagnostics({
+        code: "@azure-tools/typespec-azure-resource-manager/arm-no-record",
+      });
+  });
+
+  it("emits diagnostic when a model extends Record type", async () => {
+    await tester
+      .expect(
+        `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Contoso;
+
+      @Azure.ResourceManager.tenantResource
+      model Widget is ProxyResource<WidgetProperties> {
+        @key("widgetName")
+        @segment("widgets")
+        @path
+        @visibility("read")
+        name: string;
+      }
+
+      model WidgetProperties extends Record<string> {}
+      `
+      )
+      .toEmitDiagnostics({
+        code: "@azure-tools/typespec-azure-resource-manager/arm-no-record",
+      });
+  });
+
+  it("emits diagnostic when a model is Record type", async () => {
+    await tester
+      .expect(
+        `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Contoso;
+
+      @Azure.ResourceManager.tenantResource
+      model Widget is ProxyResource<WidgetProperties> {
+        @key("widgetName")
+        @segment("widgets")
+        @path
+        @visibility("read")
+        name: string;
+      }
```

Comment:
Do you actually need all this boilerplate for those tests?

Reply:
Yes because I updated the logic to examine only ARM resources. I refactored it to be a little cleaner.

Reply:
so if it only check arm resources is there not a gap that `bad-record` would have reported but now this doesn't?

Reply:
I don't think the azure core rule should be aware of arm at all. I think we should reapply the same logic manually?

Wouild it hurt to just complain about any reference of `Record` instead of limiting to arm resources?

Reply:
Mark's response, per Teams:

> [12:51 PM] Mark Cowlishaw
> Yeah, I don't think here is any particular admonition against Record<unknown> per se, in non-resource types.  As long as we are including the entire resource definition (properties of properties (of properties...)) then I think this is acceptable.  The main outliers will be Action request/response in that case